### PR TITLE
feat: calendar events

### PR DESCRIPTION
**Pull request by Uhuh**
Please describe the changes this PR makes and why it should be merged:

Status

-   [ ] Code changes have been tested.

Semantic versioning classification:

-   [X] This PR changes the library's interface (methods or parameters added)
-   [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
-   [ ] This PR only includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: packages/guilded-api-typings/lib/v1/structs/CalendarEvent.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    /** The duration of the event IN MINUTES (min 1) */
+    duration?: number;
+    /** If the event has a limited view */
+    isPrivate?: boolean;
+    mentions?: MentionsPayload;
+    /** The ISO 8601 timestamp that the event was created at */
+    createdAt: string;
+    /** The ID of the user who created this event */
+    createdBy: string;
+    cancellation: CalendarEventCancellationPayload;
```
Comment (zaida04) [inline packages/guilded-api-typings/lib/v1/structs/CalendarEvent.ts]:
```suggestion
    cancellation?: CalendarEventCancellationPayload;
```

**Review by zaida04 — APPROVED** [review]

**Comment by zaida04** [comment]
once you approve, go ahead and merge this @DaStormer

File: packages/rest/lib/util/Router.ts [hunk 2]
```diff
@@ -116,6 +124,31 @@ export class Router {
         return this.rest.delete<RESTDeleteChannelMessageResult>(ROUTES.channelMessage(channelId, messageId));
     }
 
+    /** Get a single calendar event. */
+    getCalendarEvent(channelId: string, calendarEventId: number): Promise<RESTGetCalendarEventResult> {
+        return this.rest.post<RESTPostCalendarEventResult>(ROUTES.calendarEvent(channelId, calendarEventId));
```
Comment (DaStormer) [inline packages/rest/lib/util/Router.ts]:
```suggestion
        return this.rest.get<RESTGetCalendarEventResult>(ROUTES.calendarEvent(channelId, calendarEventId));
```

File: packages/rest/lib/util/Router.ts [hunk 3]
```diff
@@ -116,6 +124,31 @@ export class Router {
...
+    }
+
+    /** Get all calendar events in a specific channel. */
+    getCalendarEvents(channelId: string, options: RESTGetCalendarEventsBody): Promise<RESTGetCalendarEventsResult> {
+        return this.rest.post<RESTGetCalendarEventsResult, RESTGetCalendarEventsBody>(ROUTES.calendarEvents(channelId), options);
```
Comment (DaStormer) [inline packages/rest/lib/util/Router.ts]:
```suggestion
        return this.rest.get<RESTGetCalendarEventsResult, RESTGetCalendarEventsBody>(ROUTES.calendarEvents(channelId), options);
```

File: packages/rest/lib/util/Router.ts [hunk 4]
```diff
@@ -116,6 +124,31 @@ export class Router {
...
+    }
+
+    /** Create a calendar event. */
+    createCalendarEvent(channelId: string, options: RESTPostCalendarEventBody): Promise<RESTPostCalendarEventResult> {
+        return this.rest.post<RESTPostCalendarEventResult, RESTPostCalendarEventBody>(ROUTES.calendarEvents(channelId), options);
+    }
+
+    /** Update an existing calendar event. */
+    updateCalendarEvent(channelId: string, calendarEventId: number, options: RESTPatchCalendarEventBody): Promise<RESTPatchCalendarEventResult> {
+        return this.rest.post<RESTPatchCalendarEventResult, RESTPatchCalendarEventBody>(ROUTES.calendarEvent(channelId, calendarEventId), options);
```
Comment (DaStormer) [inline packages/rest/lib/util/Router.ts]:
```suggestion
        return this.rest.patch<RESTPatchCalendarEventResult, RESTPatchCalendarEventBody>(ROUTES.calendarEvent(channelId, calendarEventId), options);
```

File: packages/rest/lib/util/Router.ts [hunk 5]
```diff
@@ -116,6 +124,31 @@ export class Router {
...
+    }
+
+    /** Delete a calendar event. */
+    deleteCalendarEvent(channelId: string, calendarEventId: number): Promise<RESTDeleteCalendarEventResult> {
+        return this.rest.post<RESTDeleteCalendarEventResult>(ROUTES.calendarEvent(channelId, calendarEventId));
```
Comment (DaStormer) [inline packages/rest/lib/util/Router.ts]:
```suggestion
        return this.rest.delete<RESTDeleteCalendarEventResult>(ROUTES.calendarEvent(channelId, calendarEventId));
```

**Review by DaStormer — CHANGES_REQUESTED** [review]

**Review by DaStormer — APPROVED** [review]

**Pull request merged**